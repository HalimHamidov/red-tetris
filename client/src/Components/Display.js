import React from "react"
import { Styledisplay, StyledTitle, StyledData } from "./styling/StyledDisplay"

const Display = ({ gameover, title, data }) => (
    <Styledisplay gameOver={gameover}>
        <StyledTitle>{title}</StyledTitle>
        <StyledData>{data}</StyledData>
    </Styledisplay>
)

export default Display;