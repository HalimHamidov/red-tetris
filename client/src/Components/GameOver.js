import React from "react";
import url from "../img/Gameover.jpeg";
import url1 from "../img/Tetris-Founder.png";
import { StyledOverlay, StyledOverlayText } from "./styling/StyledStage";

export default function GameOver({ player }) {
  return (
    <StyledOverlay>
      <StyledOverlayText>
        {player?.lost === true ? (
          <img data-testid="gameOver-Lost-img" style={{ width: "255px" }} src={url} />
        ) : (
          <img data-testid="gameOver-Won-img" style={{ width: "255px" }} src={url1} />
        )}
        <h1>{player?.lost === true ? "You lost! Try HARD!" : "You Win! Congratulations! "}</h1>
        <p>
          {player?.admin === true
            ? "Restart the game"
            : "Wait for host to restart the game"}
        </p>
      </StyledOverlayText>
    </StyledOverlay>
  );
}
