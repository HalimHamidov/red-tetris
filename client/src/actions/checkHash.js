import { toast } from "react-toastify";
import { socket } from "../hooks";
import store from "../Store";

export const HashFormat = (hash) => {
  const regexp = /(?<room>[a-zA-Z0-9]{1,18})\[(?<usr>[a-zA-Z0-9]{1,18})\]/;
  const found = hash.match(regexp);
  if (!found) {
    toast.error(
      "[!] Please Enter Up to 18 characters[!]"
    );
    return false;
  }
  return found;
};
export const checkHash = (hash) => {
  if (!hash.trim()) 
    return false;
  const found = HashFormat(hash)
  if (found){
    socket.emit("joinRoom", {
      user: found.groups.usr,
      room: found.groups.room,
      mode: store.getState().room.mode,
    });
    return true
  }
};