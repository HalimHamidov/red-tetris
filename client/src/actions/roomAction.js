export const START_GAME = "START_GAME";
export const RESTART_GAME = "RESTART_GAME";
export const GAME_OVER = "GAME_OVER";

export const ROOM_JOINED = "ROOM_JOINED";
export const UPDATE_MODE = "UPDATE_MODE";

export const CHANGE_PIECE = "CHANGE_PIECE";
export const CLEAR_PIECES = "CLEAR_PIECES";

export const UPDATE_MEMBER = "UPDATE_MEMBER";
export const UPDATE_NAME = "UPDATE_NAME";

const startgame = (data) => ({
  type: START_GAME,
  data: data,
});

const updatemember = (data) => ({
  type: UPDATE_MEMBER,
  data: data,
});

const nextpiece = (data) => ({
  type: CHANGE_PIECE,
  data: data,
});

const updatename = (data) => ({
  type: UPDATE_NAME,
  data: data,
});
export { startgame, updatemember, nextpiece, updatename };
