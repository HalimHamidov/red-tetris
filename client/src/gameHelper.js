import { Tetrominos } from "./tetrominos";
export const S_WIDTH = 10;
export const S_HEIGHT = 20;


export const Createstage = (h = S_HEIGHT, w = S_WIDTH) =>
  Array.from(Array(h), () => new Array(w).fill([0, "clear"]));

export const checkcollision = (player, stage, { x: moveX, y: moveY }) => {
  for (let y = 0; y < player.tetromino.length; y++) {
    for (let x = 0; x < player.tetromino.length; x++) {
      if (player.tetromino[y][x] !== 0) {
        if (
          !stage[y + player.pos.y + moveY] ||
          !stage[y + player.pos.y + moveY][x + player.pos.x + moveX] ||
          stage[y + player.pos.y + moveY][x + player.pos.x + moveX][1] !==
            "clear"
        )
          return true;
      }
    }
  }
  return false;
};
