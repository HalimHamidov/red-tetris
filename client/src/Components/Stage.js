import React from "react";
import Cell from "./Cell";
import { StyleStage } from "./styling/StyledStage";
import GameOver from "./GameOver";

const Stage = ({ stage, gameOver, player }) => {
  if (!stage) {
    return <span>Loading...</span>;
  }
  return (
    <StyleStage width={stage[0].length} height={stage.length}>
      {gameOver && <GameOver {...{ player }} />}
      {stage.map((row, i) =>
        row.map((cell, x) => <Cell key={x} type={cell[0]} />)
      )}
    </StyleStage>
  );
};

export default Stage;
