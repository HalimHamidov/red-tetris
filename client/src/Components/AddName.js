import React, { useState, useEffect } from "react";
import { toast } from "react-toastify";
import { addPlayerName } from "../actions/playerAction";
import { useDispatch } from "react-redux";
import { StyledButton, StyledInput, StyledAddnameWrapper } from "./styling/StyledForm";

const AddName = () => {
  const dispatch = useDispatch();
  const [username, setUserName] = useState("");

  function adduser(e) {
    e.preventDefault();
    if (username.trim() !== "") {
      dispatch(addPlayerName(username.trim()));
    }
    else {
      setUserName("")
      toast.error(
        "Please Enter correct 1 to 12 characters in length player and room names"
      );
    }
  }
  useEffect(() => {
    document.getElementById("username-i") && document.getElementById("username-i").focus();
  });
  return (
    <StyledAddnameWrapper  onSubmit={(e) => adduser(e)}>
      <StyledInput
        id="username-i"
        placeholder="Enter Username"
        onChange={(e) => setUserName(e.target.value)}
      />
      <StyledButton>Submit</StyledButton>
    </StyledAddnameWrapper>
  );
};

export default AddName;
