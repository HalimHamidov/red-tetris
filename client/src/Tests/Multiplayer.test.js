import renderer from "react-test-renderer";
import Multiplayer from "../Components/Multiplayer";
import store from "../Store";
import { Provider } from "react-redux";
import rRoom from "../reducers/roomReducer";
import * as aRoom from "../actions/roomAction";
import { checkcollision } from "../gameHelper";
import {updatePlayerPos} from "../Components/Multiplayer";
test("Multiplayer render test ", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Multiplayer />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});


