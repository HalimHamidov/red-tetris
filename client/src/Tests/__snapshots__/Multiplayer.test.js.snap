// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiplayer render test  1`] = `
<div
  className="sc-jrQzAO lhRSqT"
  data-testid="styled-tet-wra"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="container"
  >
    <div
      className="sc-kDTinF jkRuxW"
    >
      <div
        className="w-100 row"
      >
        <div
          className="flex justify-center red-tetris__col-stage mb-20 col-lg-12 col-xl-12"
        >
          <div
            className="sc-gsDKAQ bzNBex"
            height={20}
            width={10}
          >
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
            <div
              className="sc-bdvvtL yBcMv"
              color="0, 0, 0"
              type={0}
            />
          </div>
          <aside>
            <div
              className="sc-gKclnd jyHzmI"
            >
              <div
                className="sc-iCfMLu emHtQj"
              >
                Next
              </div>
              <div
                className="sc-pVTFL VYhPp"
              />
            </div>
            <div
              className="sc-jRQBWg jDBOnt"
            >
              <div
                className="sc-iCfMLu emHtQj"
              >
                Score
              </div>
              <div
                className="sc-furwcr fMMMyD"
              >
                0
              </div>
            </div>
            <div
              className="sc-jRQBWg jDBOnt"
            >
              <div
                className="sc-iCfMLu emHtQj"
              >
                Level
              </div>
              <div
                className="sc-furwcr fMMMyD"
              >
                0
              </div>
            </div>
            <div
              className="sc-jRQBWg jDBOnt"
            >
              <div
                className="sc-iCfMLu emHtQj"
              >
                Rows
              </div>
              <div
                className="sc-furwcr fMMMyD"
              >
                0
              </div>
            </div>
            <div
              className="sound-icon-wrapper"
              style={
                Object {
                  "cursor": "pointer",
                  "padding": "10px 0px 20px",
                  "textAlign": "center",
                }
              }
            >
              
              
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-volume-mute fa-w-16 fa-2x sound-icon"
                data-icon="volume-mute"
                data-prefix="fas"
                focusable="false"
                onClick={[Function]}
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M215.03 71.05L126.06 160H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V88.02c0-21.46-25.96-31.98-40.97-16.97zM461.64 256l45.64-45.64c6.3-6.3 6.3-16.52 0-22.82l-22.82-22.82c-6.3-6.3-16.52-6.3-22.82 0L416 210.36l-45.64-45.64c-6.3-6.3-16.52-6.3-22.82 0l-22.82 22.82c-6.3 6.3-6.3 16.52 0 22.82L370.36 256l-45.63 45.63c-6.3 6.3-6.3 16.52 0 22.82l22.82 22.82c6.3 6.3 16.52 6.3 22.82 0L416 301.64l45.64 45.64c6.3 6.3 16.52 6.3 22.82 0l22.82-22.82c6.3-6.3 6.3-16.52 0-22.82L461.64 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </aside>
        </div>
        <div
          className="spectersWrapper col-lg-12 col-xl-5"
        >
          <div
            className="w-100 row"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
